### DISCUSSION TITLE
esp wifi connect fussiness

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#12444] [Hardware & Peripherals] [answered]
My PC does so much better at connecting to wifi than my esp32! 

Recently I spent a whole day trying to figure out why the PC was able to connect to an open AP when the esp32 wouldn't. Turned out that pwd in sta.connect(AP, pwd) has to be None if you want the esp32 to connect to an open AP.

Today I've got an AP that is temporarily not plugged into the router LAN. My PC happily connects to this AP with a polite msg 'no internet access' but the esp32 won't. When I jack the AP into the LAN the esp32 will connect.

I'm having a much harder time figuring this out. Can anyone suggest why lack of internet access on the AP should prevent the esp32 from connecting to it?

**Comment by ctimmer** [comment, marked as answer]
The esp32 may be looking for a gateway and DNS address. Try using ifconfig to set these manually with a static IP address.
Curt

**Reply by kjm1102** [reply]
sta.ifconfig(('10.0.0.2', '255.255.255.0', '10.0.0.1', '8.8.8.8')) cures the connect refusal nicely Curt, tnks heaps
